float32 value
float32 velocity